Automattic/wp-calypso
# Launchpad: Add launchpad navigator to masterbar

Related to #

## Proposed Changes

* This PR adds the core logic needed to show users a task list in the masterbar, though the functionality is protected by the `launchpad/navigator` feature flag, which is disabled in all environments by default.
* The functionality for the content will be built in future PRs, as will the changes to use a dedicated data store for the modal's state. There will also be follow-up work to show the content in other contexts like the editor and wp-admin.
* At present, we don't handle cases where no site is selected, or cases where the current site doesn't have an active checklist

## Testing Instructions

* Run this branch via Calypso.live as per the link in [this comment](https://github.com/Automattic/wp-calypso/pull/81652#issuecomment-1716318211).
* Navigate to a site that is showing a task list in Customer Home, either after skipping the task list or because we have a post-launch task list for that site
* Verify that we _don't_ show the task list icon in the master bar
* Now add `?flags=launchpad/navigator` to the URL
* Verify that you _do_ see the task list icon in the master bar
* Click on the icon
* Verify that we show tasks from your current task list in the popup

## Pre-merge Checklist



- [N/A] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [N/A] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [N/A] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
